src/wp-includes/media.php

```diff
@@ -774,24 +774,28 @@ function image_get_intermediate_size( $post_id, $size = 'thumbnail' ) {
 			$imagedata['width']  = $imagedata['sizes']['full']['width'];
 		}
 
-		foreach ( $imagedata['sizes'] as $_size => $data ) {
+		foreach ( $imagedata['sizes'] as $_size => $size_data ) {
+			if ( ! is_array( $size_data ) || ! isset( $size_data['width'] ) || ! isset( $size_data['height'] ) ) {
+				continue;
```

Comment:
Silently skipping isn't preferred here. Why? 

`wp_get_attachment_metadata()` returns an array on success and `'sizes'` is expected to be an `array`. If it's not an array, then something went wrong. Silently skipping makes it harder for developers to debug what happened and could lead to confusion for users.